File: system-test/java/src/test/java/com/optivem/eshop/systemtest/_legacy/mod08/e2e/PlaceOrderPositiveTest.java
package com.optivem.eshop.systemtest._legacy.mod08.e2e;

import com.optivem.eshop.dsl.channel.ChannelType;
import com.optivem.eshop.dsl.driver.port.shop.dtos.OrderStatus;
import com.optivem.eshop.systemtest._legacy.mod08.e2e.base.BaseE2eTest;
import com.optivem.testing.Channel;
import org.junit.jupiter.api.TestTemplate;

class PlaceOrderPositiveTest extends BaseE2eTest {
    @TestTemplate
    @Channel({ChannelType.UI, ChannelType.API})
    void shouldPlaceOrder() {
        scenario
                .given().product().withUnitPrice(20.00)
                .when().placeOrder().withQuantity(5)
                .then().shouldSucceed()
                .and().order()
                .hasOrderNumberPrefix("ORD-")
                .hasQuantity(5)
                .hasUnitPrice(20.00)
                .hasTotalPriceGreaterThanZero()
                .hasStatus(OrderStatus.PLACED);
    }
}

File: system-test/dotnet/SystemTests/Legacy/Mod08/E2eTests/PlaceOrderPositiveTest.cs
using Dsl.Core.Shop;
using Driver.Port.Shop.Dtos;
using SystemTests.Legacy.Mod08.E2eTests.Base;
using Optivem.Testing;
using Xunit;

namespace SystemTests.Legacy.Mod08.E2eTests;

public class PlaceOrderPositiveTest : BaseE2eTest
{
    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task ShouldPlaceOrderWithCorrectSubtotalPrice(Channel channel)
    {
        await Scenario(channel)
            .Given().Product()
                .WithUnitPrice("20.00")
            .When().PlaceOrder()
                .WithQuantity(5)
            .Then().ShouldSucceed()
            .And().Order()
                    .HasTotalPrice("100.00");
    }

    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    [ChannelInlineData("20.00", "5", "100.00")]
    [ChannelInlineData("10.00", "3", "30.00")]
    [ChannelInlineData("15.50", "4", "62.00")]
    [ChannelInlineData("99.99", "1", "99.99")]
    public async Task ShouldPlaceOrderWithCorrectSubtotalPriceParameterized(Channel channel, string unitPrice, string quantity, string subtotalPrice)
    {
        await Scenario(channel)
            .Given().Product()
                .WithUnitPrice(unitPrice)
            .When().PlaceOrder()
                .WithQuantity(quantity)
            .Then().ShouldSucceed()
            .And().Order()
                    .HasTotalPrice(subtotalPrice);
    }

    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task ShouldPlaceOrder(Channel channel)
    {
        await Scenario(channel)
            .Given().Product()
                .WithUnitPrice("20.00")
            .When().PlaceOrder()
                .WithQuantity(5)
            .Then().ShouldSucceed()
            .And().Order()
                    .HasOrderNumberPrefix("ORD-")
                    .HasQuantity(5)
                    .HasUnitPrice(20.00m)
                    .HasTotalPrice("100.00")
                    .HasStatus(OrderStatus.Placed);
    }
}













